classes/class-nets-easy-ajax.php

```diff
@@ -147,14 +147,18 @@ public static function get_order_data() {
 		}
 
 		$payment_id = filter_input( INPUT_POST, 'paymentId', FILTER_SANITIZE_FULL_SPECIAL_CHARS );
-		if ( ! $payment_id ) {
-			$payment_id = WC()->session->get( 'dibs_payment_id' );
-		}
 
 		if ( empty( $payment_id ) ) {
 			wp_send_json_error( 'Failed to get the Nexi order data. Payment ID missing.' );
 		}
 
+		$payment_id_session = WC()->session->get( 'dibs_payment_id' );
+
+		if ( $payment_id !== $payment_id_session ) {
+			wp_send_json_error( 'Failed to get the Nexi order data. Payment ID does not match the session.' );
```

Comment:
`wp_send_json_error` will trigger the `wp_die` or die function, so the row after this will never happen. You should put this last in the if statement